instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/engine.py

```diff
@@ -163,5 +163,7 @@ def _get_attributes_from_cursor(vendor, cursor, attrs):
                 data = parse_dsn(dsn)
                 attrs[SpanAttributes.DB_NAME] = data.get("dbname")
                 attrs[SpanAttributes.NET_PEER_NAME] = data.get("host")
-                attrs[SpanAttributes.NET_PEER_PORT] = int(data.get("port"))
+                # parse_dsn may omit port when connecting via unix socket
+                if data.get("port"):
+                    attrs[SpanAttributes.NET_PEER_PORT] = int(data["port"])
```

Comment:
If it is a unix socket path we can add different span attributes 
https://github.com/open-telemetry/opentelemetry-python-contrib/blob/fbb677a01d82b6e4602bed5a555ee38474cd9046/instrumentation/opentelemetry-instrumentation-redis/src/opentelemetry/instrumentation/redis/util.py#L44-L47.

Reply:
Makes sense. Let's add NET_PEER_TRANSPORT conditionally when connecting via unix socket.

Reply:
@owais that's a great point.

It looks like the `TypeError` could apply to non-socket cases too.

```py
from psycopg2.extensions import parse_dsn
from sqlalchemy import create_engine

engine = create_engine('postgresql://postgres@localhost/postgres')
conn = engine.raw_connection()
cur = conn.cursor()

cur.connection.dsn  # 'host=localhost user=postgres dbname=postgres
parse_dsn(cur.connection.dsn)  # {'user': 'postgres', 'dbname': 'postgres', 'host': 'localhost'}
```

However, there is `cur.connection.info` which provides:

```
cur.connection.info.port  # 5432
cur.connection.info.host  # 'localhost'
```

When connecting to a socket it provides:

```
cur.connection.info.port  # 5432
cur.connection.info.host  # '/var/run/postgresql/'
``

`psycopg2` makes the assumption that every socket is named `.s.PGSQL.{port}`, meaning it only stores the absolute path of the socket directory. Mirroring that assumption here seems risky.

The `info` object seems like the safer bet since it's always provides certain values.
If you initialize a SQLAlchemy engine with `engine = create_engine('postgresql://')` the `.dsn` attribute will be an empty string.

I think at the very least the presence of a filesystem path in `info.host` is a clue we are using sockets... but I'll dig around the `psycopg2` source a bit to see if I can find any better guarantees.

Reply:
Makes sense. Checking if the host starts with `unix://`, `unix:///` or `/` makes more sense to me than checking for absence of port.